web/src/test/java/org/springframework/security/web/webauthn/registration/PublicKeyCredentialCreationOptionsFilterTests.java

```diff
@@ -68,11 +72,40 @@ class PublicKeyCredentialCreationOptionsFilterTests {
 	@Mock
 	private WebAuthnRelyingPartyOperations rpOperations;
 
+	private PublicKeyCredentialCreationOptionsFilter filter;
+
+	private MockHttpServletRequest request;
+
+	private MockHttpServletResponse response;
+
+	private FilterChain chain;
+
+	@BeforeEach
+	public void setup() {
+		this.filter = new PublicKeyCredentialCreationOptionsFilter(this.rpOperations);
+		this.request = new MockHttpServletRequest();
+		this.response = new MockHttpServletResponse();
+		this.chain = mock(FilterChain.class);
+	}
+
 	@AfterEach
 	void clear() {
 		SecurityContextHolder.clearContext();
 	}
 
+	@Test
+	public void doFilterWhenCustomRequestMatcherThenUses() throws Exception {
+		this.request.setPathInfo("/path");
+		this.filter.setRequestMatcher(new AntPathRequestMatcher("/path"));
```

Comment:
Please create a mock `RequestMatcher` instead; the current test doesn't necessarily confirm that the request matcher was used to cause no filter chain interactions. If you use a mock `RequestMatcher`, you can verify that it was interacted with. This will make the test setup simpler as well:

```java
RequestMatcher requestMatcher = mock(RequestMatcher.class);
this.filter.setRequestMatcher(requestMatcher);
this.filter.doFilter(null, null, null);
verfiy(requestMatcher).matches(any());
```

Reply:
Note that I think a behavioral test can also be helpful; we just want to make sure that we at least have a test that guarantees that the request matcher is invoked.